src/kernels.h

```diff
@@ -0,0 +1,15 @@
+#ifndef KERNELS_H_
+#define KERNELS_H_
+#include <stdio.h>
+
+//main methods
+void kernel_square_histogram(const unsigned *dim, const float *kernel, float *blurred_histogram, const float *histogram);
+void kernel_gauss_histogram(const unsigned *dim, const float *kernel, float *blurred_histogram, const float *histogram);
+void kernel_circle_rgbwt(const unsigned *dim, const float *radius, float *blurred_RGBWT, const float *RGBWT);
+void kernel_gauss_rgbwt(const unsigned *dim, const float *kernel, float *blurred_RGBWT, const float *RGBWT);
+
+
+//helper methods
+float apply_kernel(const float *kernel, const float *histogram, const unsigned *dim, const size_t x, const size_t y);
+
+#endif
```

Comment:
missing eol (this can be fixed en masse from shell, like:
```sh
find -name '*.c' -or -name '*.h' -or -name '*.R' | while read filename ; do
    # remove any trailing spaces
    sed --in-place -e 's/\s*$//' "$filename"
    # add a final newline if missing
    if [[ -s "$filename" && $(tail -c 1 "$filename" |wc -l) -eq 0 ]]
    then echo >> "$filename"
    fi
    # squash superfluous final newlines
    sed -i -e :a -e '/^\n*$/{$d;N;};/\n$/ba' "$filename"
done
```
)